app/dao/fact_notification_status_dao.py

```diff
@@ -246,7 +246,18 @@ def fetch_notification_status_for_service_for_today_and_7_previous_days(service_
         FactNotificationStatus.notification_type.label("notification_type"),
         FactNotificationStatus.notification_status.label("status"),
         *([FactNotificationStatus.template_id.label("template_id")] if by_template else []),
-        FactNotificationStatus.notification_count.label("count"),
+        *(
+            [
+                case(
+                    [
+                        (FactNotificationStatus.notification_type == "email", FactNotificationStatus.notification_count),
+                    ],
+                    else_=FactNotificationStatus.billable_units,
+                ).label("count")
+            ]
+            if current_app.config["FF_SPIKE_SMS_DAILY_LIMIT"]
+            else [FactNotificationStatus.notification_count.label("count")]
+        ),
```

Comment:
This creates a SQL case statement to choose either `notification_count` or `billable_units`, depending on `notification_type`, roughly this in SQL:
```
SELECT notification_type, notification_status, template_id
  CASE 
    WHEN notification_type = 'email' THEN notification_count
    ELSE billable_units
  END AS count,
FROM ft_notification_status
```

Reply:
Yes! It is definitely strange looking syntax, but that is the api that [SqlAlchemy provides for a case statement](https://docs.sqlalchemy.org/en/14/core/sqlelement.html#sqlalchemy.sql.expression.case)!